src/utils/codec.ts

```diff
@@ -0,0 +1,318 @@
+/* eslint-disable no-bitwise */
+/**
+ * @file Codec
+ * @name codec.ts<utils>
+ * @description |
+ *
+ *    |    base    |  bitwise  |  bits  |
+ *    | ---------- | --------- | ------ |
+ *    |    16      |  1 << 4   |   4    |
+ *    |    64      |  1 << 6   |   6    |
+ *    |    256     |  1 << 8   |   8    |
+ *    |    1024    |  1 << 10  |   10   |
+ *
+ */
+interface CodecScheme {
+    bits: number
+    table: string[]
+    tail: string
+}
+
+/**
+ * Binary Encoding
+ */
+// prettier-ignore
+export const Binary: CodecScheme = {
+    bits: 1,
+    table: [
+        "0",
+        "1",
+    ],
+    tail: "",
+}
+
+/**
+ * Hex Encoding
+ */
+// prettier-ignore
+export const HEX: CodecScheme = {
+    bits: 4,
+    table: [
+        "0", "1", "2", "3",
+        "4", "5", "6", "7",
+        "8", "9", "a", "b",
+        "c", "d", "e", "f",
+    ],
+    tail: "",
+}
+
+/**
+ * Base64 Encoding
+ */
+// prettier-ignore
+export const BASE64: CodecScheme = {
+    bits: 6,
+    table: [
```

Comment:
Can be designed as a function.
```typescript
toPoint(index: number) {
    let point: number
    if (index < 26) {
        point = 0x40 + index
    } else if (index < 52) {
        point = 0x60 + index
    } else if (index === 52) {
        point = 0x2B
    } else if (index === 53) {
        point = 0x2F
    }
    return point
} 
```

Reply:
I don't think `function` is better than the static string array:

+ Hard to read
+ Calculating costs performances
+ What if coding schemes are not in order?

Reply:
No long string no too many `if-else` and the costs could be ignored,  if we selecting ranges of emojis by unicode coding, I agree with using `function` instead.

Reply:
And I prefer to use `String.fromCodePoint(0x1f980)` instead of `"🦀"` actually...

Reply:
It might be better to format the `base1024.ts` as `base2048.ts`